### PR TITLE
replace pcap dependency with homegrown solution

### PR DESCRIPTION
**Pull request by cpacejo**
The major change here is adding the `pcap_writer` module, which just implements a trivially simple PCAP file writer.  The key differences from the `pcap` crate or `pcap_file` are:
1. It's asynchronous – this means if the PCAP destination is e.g. a network file system or an SSH tunnel, it won't block the Tokio runtime.
2. It's not `pcap` – meaning we now can eliminate that (and the big C library which it depends on) as a dependency from the PH.

The remainder of changes just switch to using this instead of `pcap`, which, beside type changes, mostly are (1) switching some methods to async, (2) handling some new errors, and (3) opening the `File` ourselves.

(Long-term plan is also to switch away from having a `File` as the backing store, and instead allowing an arbitrary file descriptor, which is passed in via the RPC mechanism (#176).)

### CODE REVIEW

File: adapter/ph/src/capture_worker.rs [hunk 1]
```diff
@@ -1,46 +1,50 @@
...
+            Some(PcapWriter::open(File::open(path).await?, linktype::USER0).await?);
+        Ok(())
     }
 
-    pub async fn flush_capture_file(&self) -> Result<(), Error> {
+    pub async fn flush_capture_file(&self) -> Result<(), io::Error> {
         let sf = &mut self.inner_cap.lock().await.savefile;
         match sf {
-            Some(ref mut sf) => sf.flush(),
+            Some(ref mut sf) => sf.flush().await,
```
Comment (owenrhysthomas) [inline adapter/ph/src/capture_worker.rs:35]: do you want a `?` here too? Otherwise I don't see how this fn can return an `io::Error`

**Review by owenrhysthomas — APPROVED** [review]
Nice! Just the one question about the `?`

File: adapter/ph/src/capture_worker.rs (hunk 1 above)
Comment (cpacejo) [reply adapter/ph/src/capture_worker.rs:35]: `sf.flush().await` itself returns a `Result<(), io::Error>` – so here I'm just returning directly whatever it returns.  (I could alternatively write `{ sf.flush().await?; Ok(()) }` though.)

Comment (owenrhysthomas) [reply adapter/ph/src/capture_worker.rs:35]: Oh, I see. That makes sense.

**Pull request merged**